server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/VacationRoutesTest.java

```diff
@@ -0,0 +1,297 @@
+        given()
+            .body("{\"enabled\":true,\"subject\":\"On vacation again\"}")
+        .when()
+            .post(VacationRoutes.VACATION + SEPARATOR + CAROL)
+        .then()
+            .statusCode(HttpStatus.BAD_REQUEST_400);
+    }
+
+    @Test
+    void patchVacationCreates() throws Exception {
+        AccountId bob = AccountId.fromString(BOB);
+        RecipientId alice = RecipientId.fromMailAddress(new MailAddress(ALICE));
+
+        vacationRepository.modifyVacation(bob, VacationPatch.builderFrom(VACATION).build())
+            .then(notificationRegistry.register(bob, alice, Optional.empty())).block();
+
+        given()
+            .body("{\"enabled\":true,\"fromDate\":\"2021-09-20T10:00:00Z\",\"toDate\":\"2021-09-27T18:00:00Z\"," +
+                "\"subject\":\"On vacation again\",\"textBody\":\"Need more vacation!\",\"htmlBody\":\"<p>Need more vacation!</p>\"}")
+        .when()
+            .patch(VacationRoutes.VACATION + SEPARATOR + BOB)
+        .then()
+            .statusCode(HttpStatus.NO_CONTENT_204);
+
+        Vacation vacation = vacationRepository.retrieveVacation(bob).block();
+        assertThat(vacation).isNotNull();
+        assertThat(vacation.isEnabled()).isTrue();
+        assertThat(vacation.getFromDate()).isEqualTo(Optional.of(ZonedDateTime.parse("2021-09-20T10:00:00Z[UTC]")));
+        assertThat(vacation.getToDate()).isEqualTo(Optional.of(ZonedDateTime.parse("2021-09-27T18:00:00Z[UTC]")));
+        assertThat(vacation.getSubject()).isEqualTo(Optional.of("On vacation again"));
+        assertThat(vacation.getTextBody()).isEqualTo(Optional.of("Need more vacation!"));
+        assertThat(vacation.getHtmlBody()).isEqualTo(Optional.of("<p>Need more vacation!</p>"));
+
+        Boolean registered = notificationRegistry.isRegistered(bob, alice).block();
+        assertThat(registered).isFalse();
+    }
+
+    @Test
+    void patchVacationModifies() throws Exception {
+        AccountId bob = AccountId.fromString(BOB);
+        RecipientId alice = RecipientId.fromMailAddress(new MailAddress(ALICE));
+
+        vacationRepository.modifyVacation(bob, VacationPatch.builderFrom(VACATION).build())
+            .then(notificationRegistry.register(bob, alice, Optional.empty())).block();
+
+        given()
+            .body("{\"enabled\":true,\"subject\":\"More vacation\"}")
+        .when()
+            .patch(VacationRoutes.VACATION + SEPARATOR + BOB)
+        .then()
+            .statusCode(HttpStatus.NO_CONTENT_204);
+
+        Vacation vacation = vacationRepository.retrieveVacation(bob).block();
+        assertThat(vacation).isNotNull();
+        assertThat(vacation.isEnabled()).isTrue();
+        assertThat(vacation.getFromDate()).isEmpty();
+        assertThat(vacation.getToDate()).isEmpty();
+        assertThat(vacation.getSubject()).isEqualTo(Optional.of("More vacation"));
+        assertThat(vacation.getTextBody()).isEmpty();
+        assertThat(vacation.getHtmlBody()).isEmpty();
```

Comment:
In `patchVacationModifies`

I would expect only the subject to be changed, but the other properties to be untouched.

Reply:
PATCH was supposed to reset any properties missing/null in the request JSON. But since I removed PATCH, this is a moot point :-)